### PR TITLE
Facets

### PR DESCRIPTION
**Pull request by MarkShleifer**

### CODE REVIEW

File: src/app/packages/menu-items/cases/models/case.model.ts [hunk 1]
```diff
@@ -22,11 +22,9 @@ export type CaseTimeState = {
 	to: string
 }
 export type CaseFacetsState = {
-	SensorName: string,
-	SensorType: 'SAR' | 'VIS' | 'IR' | 'Satellite',
-	Stereo: boolean,
-	Resolution: number
+	filters: [ {fieldName: string, metadata: any}]
```
Comment (yairtawil) [inline src/app/packages/menu-items/cases/models/case.model.ts]: change to {fieldName: string, metadata: any} []

File: src/app/packages/menu-items/filters/components/enum-filter-container/enum-filter-container.component.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      </div>
+      <span class="only-filter" (click)="selectOnly(enumMetadata.key)">only</span>
+    </div>
+  </div>
+
+  <div class="right-side">
+    <span>{{enumMetadata.value.count}}</span>
+  </div>
+
+</div>
```
Comment (yairtawil) [inline src/app/packages/menu-items/filters/components/enum-filter-container/enum-filter-container.component.html]: EOF

File: src/app/packages/menu-items/filters/components/filter-container/filter-container.component.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  ]
+})
+export class FilterContainerComponent {
+
+  @Input() filter: Filter;
+
+  @ViewChild('fields') fields: ElementRef;
+
+  private show = true;
+  private _isAnimatig = false;
```
Comment (yairtawil) [inline src/app/packages/menu-items/filters/components/filter-container/filter-container.component.ts]: delete

File: src/app/packages/menu-items/filters/components/filter-container/filter-container.component.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  showAll(): void {
+    if (this.metadataFromState) {
+      const clonedMetadata: FilterMetadata = Object.assign(Object.create(this.metadataFromState), this.metadataFromState);
+      clonedMetadata.showAll();
+
+      this.onMetadataChange(clonedMetadata);
+    }
+  }
+}
```
Comment (yairtawil) [inline src/app/packages/menu-items/filters/components/filter-container/filter-container.component.ts]: EOF

File: src/app/packages/menu-items/filters/components/filters-collection/filters-collection.component.html [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<div class="collection">
+  
+  <ng-template ngFor let-filter [ngForOf]="initialFilters$ | async">
+    <ansyn-filter-container [filter]="filter"></ansyn-filter-container>
+  </ng-template>
+  
+</div>
```
Comment (yairtawil) [inline src/app/packages/menu-items/filters/components/filters-collection/filters-collection.component.html]: EOF

File: src/app/packages/menu-items/filters/components/filters-collection/filters-collection.component.less [hunk 6]
```diff
@@ -0,0 +1,3 @@
+.collection{
+    padding: 25px 10px;
+}
```
Comment (yairtawil) [inline src/app/packages/menu-items/filters/components/filters-collection/filters-collection.component.less]: EOF

File: src/app/packages/menu-items/filters/models/metadata/enum-filter-metadata.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        this.enumsFields = new Map<string, { count: number, isChecked: boolean }>();
+        this.type = 'Enum';
+    }
+
+    updateMetadata(key: string): void {
+        this.enumsFields.get(key).isChecked = !this.enumsFields.get(key).isChecked;
+    }
+
+    selectOnly(selectedKey: string): void {
+        Array.from(this.enumsFields.keys()).forEach((key: string) => {
```
Comment (yairtawil) [inline src/app/packages/menu-items/filters/models/metadata/enum-filter-metadata.ts]: try enumsFields.forEach

File: src/app/app-reducers/effects/filters.app.effects.ts [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+            actionsArray.push(new UpdateCaseAction(selected_case));
+
+            return Observable.from(actionsArray);
+        })
+        .share();
+
+    @Effect()
+    resetFilters$: Observable<any> = this.actions$
+        .ofType(CasesActionTypes.SELECT_CASE_BY_ID)
```
Comment (yairtawil) [inline src/app/app-reducers/effects/filters.app.effects.ts]: it's better to depend on LOAD_OVERLAYS and LOAD_OVERLAYS_SUCCESS

File: src/app/packages/overlays/services/overlays.service.ts [hunk 9]
```diff
@@ -60,22 +60,22 @@ export class OverlaysService {
...
-            //maybe I will do it inside the event drops component itself
-            //so the component will be able to get 2 types Array and Map
-
-            overlays.forEach(value => overlaysData.push({ id: value.id, date: value.date }));
 
-            result.push({
-                name: undefined,
-                data: overlaysData
+        if (!filters || !Array.isArray(filters)) {
+            overlays.forEach(overlay => overlaysData.push({ id: overlay.id, date: overlay.date }));
```
Comment (yairtawil) [inline src/app/packages/overlays/services/overlays.service.ts]: use map instead

File: src/app/packages/overlays/services/overlays.service.ts [hunk 10]
```diff
@@ -60,22 +60,22 @@ export class OverlaysService {
...
+        } else {
+            overlays.forEach(overlay => {
+                if (filters.every(filter => filter.filterFunc(overlay, filter.filteringParams))) {
```
Comment (yairtawil) [inline src/app/packages/overlays/services/overlays.service.ts]: use filter and map

Comment (yairtawil) [inline src/app/packages/overlays/services/overlays.service.ts]: wowww

Comment (MarkShleifer) [inline src/app/packages/overlays/services/overlays.service.ts]: 10x

Comment (MarkShleifer) [inline src/app/packages/overlays/services/overlays.service.ts]: no filter and map in MAP

File: src/app/packages/menu-items/filters/components/filters-collection/filters-collection.component.less (hunk 6 above)
Comment (MarkShleifer) [inline src/app/packages/menu-items/filters/components/filters-collection/filters-collection.component.less]: done

File: src/app/packages/menu-items/filters/components/filters-collection/filters-collection.component.html (hunk 5 above)
Comment (MarkShleifer) [inline src/app/packages/menu-items/filters/components/filters-collection/filters-collection.component.html]: done

File: src/app/packages/menu-items/filters/components/filter-container/filter-container.component.ts (hunk 4 above)
Comment (MarkShleifer) [inline src/app/packages/menu-items/filters/components/filter-container/filter-container.component.ts]: done

File: src/app/packages/menu-items/filters/components/filter-container/filter-container.component.ts (hunk 3 above)
Comment (MarkShleifer) [inline src/app/packages/menu-items/filters/components/filter-container/filter-container.component.ts]: done

File: src/app/packages/menu-items/filters/components/enum-filter-container/enum-filter-container.component.html (hunk 2 above)
Comment (MarkShleifer) [inline src/app/packages/menu-items/filters/components/enum-filter-container/enum-filter-container.component.html]: done

File: src/app/packages/menu-items/cases/models/case.model.ts (hunk 1 above)
Comment (MarkShleifer) [inline src/app/packages/menu-items/cases/models/case.model.ts]: done

File: src/app/packages/overlays/services/overlays.service.ts (hunk 9 above)
Comment (MarkShleifer) [inline src/app/packages/overlays/services/overlays.service.ts]: no filter and map in MAP

File: src/app/app-reducers/effects/filters.app.effects.ts (hunk 8 above)
Comment (MarkShleifer) [inline src/app/app-reducers/effects/filters.app.effects.ts]: done

File: src/app/app-reducers/effects/filters.app.effects.ts [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+@Injectable()
+export class FiltersAppEffects {
+
+    @Effect()
+    initializeFilters$: Observable<any> = this.actions$
+        .ofType(OverlaysActionTypes.LOAD_OVERLAYS_SUCCESS)
+        .map(toPayload)
+        .withLatestFrom(this.store$.select('cases'))
+        .switchMap(([overlays, casesState]: [Overlay[], ICasesState]) => {
+            return this.filtersService.loadFilters().mergeMap((filters: Filter[]) => {
```
Comment (MarkShleifer) [inline src/app/app-reducers/effects/filters.app.effects.ts]: done

**Pull request merged**